### ISSUE TITLE
Wrong unused_result_ok version

### ISSUE THREAD

**Issue by TomFryersMidsummer** [rust-lang/rust-clippy#13497]
### Summary

[unused_result_ok says it was introduced in Clippy 1.70](https://github.com/rust-lang/rust-clippy/blob/a01975b15238b9ea9c1fd9c9fe446b5bb01530f4/clippy_lints/src/unused_result_ok.rs#L29), but [it's much newer than that](https://github.com/rust-lang/rust-clippy/pull/12150). In particular, it doesn’t exist in the latest stable version 1.81.

### Reproducer

I tried this code:

```rust
clippy-driver -W clippy::unused_result_ok main.rs
```

I expected to see this happen: [no error]

Instead, this happened:
```
warning[E0602]: unknown lint: `clippy::unused_result_ok`
  |
  = help: a lint with a similar name exists in `rustc` lints: `unused_results`
  = note: requested on the command line with `-W clippy::unused_result_ok`
  = note: `#[warn(unknown_lints)]` on by default
```

### Version

rustc 1.81.0 (eeb90cda1 2024-09-04)
binary: rustc
commit-hash: eeb90cda1969383f56a2637cbd3037bdf598841c
commit-date: 2024-09-04
host: x86_64-unknown-linux-gnu
release: 1.81.0
LLVM version: 18.1.7
